solutions/bronze/cf-581D.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: cf-581D
+source: CF
+title: Three Logos
+author: Chongtian Ma
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/20508)
+
+## Implementation
+
+We can brute force all rotations of the three rectangles with a bitmask of $2^3$, with bit $i$ on representing the $i$'th rectangle should be rotated by
+$90$ degrees. The rest is just checking if the configurations are valid.
+
+**Time Complexity:** $\mathcal{O}(2^n \cdot \textbf{side}^4)$, where $n$ is the number of rectangles (in this case, 3) and $\textbf{side}$ as the side length of the square.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int n = 3;
+	vector<pair<int, int>> logos(n);
+	for (int i = 0; i < n; i++) {
+		scanf("%d%d", &logos[i].first, &logos[i].second);
```

Comment:
why scanf/printf now

Reply:
yes

Reply:
what

Reply:
what

Reply:
does it make a differnce

Reply:
idk we use cin/cout like everywhere lol

Reply:
scanf users are underrepresented

Reply:
for consistency's sake, i'd rather you change it to cin/cout

Reply:
ive used scanf in other places too though